edgelet/build/linux/package.sh

```diff
@@ -183,13 +183,15 @@ case "$PACKAGE_ARCH" in
         ;;
 
     'arm32v7')
+        CMAKE_ARGS="$CMAKE_ARGS -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_SYSTEM_VERSION=1"
```

Comment:
Should we set `CMAKE_SYSTEM_VERSION` to something more accurate. i.e the target system version?

Reply:
I'll go through the `.cmake`s again and see what exactly cmake was looking for in these too to determine the libdir. Technically even before the `package.sh` refactoring we were defining `Linux/1` (see Mahesh's commit linked in the commit message), so it shouldn't be worse than what it was back then. Then again, the arm32 and aarch64 packages before the refactoring were broken because of the malformed openssl dependency anyway, so they clearly had never been tested and `Linux/1` may never have been right for CentOS at all.

I also need to check what the libdir is on centos aarch64; if it's supposed to be `/usr/lib64` then `Linux/1` will probably break it too.

Reply:
Yeah, CentOS aarch64 does use `/usr/lib64`, and `linux/1` does do the wrong thing for it. So we do need to figure out what exactly is needed to make it work.

Reply:
I have been enlightened.

`CMakeDetermineSystem.cmake`

>```
># if CMAKE_SYSTEM_NAME is here already set, either it comes from a toolchain file
># or it was set via -DCMAKE_SYSTEM_NAME=...
># if that's the case, assume we are crosscompiling
>if(CMAKE_SYSTEM_NAME)
>  if(NOT DEFINED CMAKE_CROSSCOMPILING)
>    set(CMAKE_CROSSCOMPILING TRUE)
>  endif()
>else()
>  set(CMAKE_SYSTEM_NAME      "${CMAKE_HOST_SYSTEM_NAME}")
>  set(CMAKE_SYSTEM_VERSION   "${CMAKE_HOST_SYSTEM_VERSION}")
>  set(CMAKE_SYSTEM_PROCESSOR "${CMAKE_HOST_SYSTEM_PROCESSOR}")
>  set(CMAKE_CROSSCOMPILING FALSE)
>endif()
>```

So it should also not be defined for the CentOS arm32 and aarch64 targets since those are being compiled natively. I confirmed that an aarch64 build on my machine with the two not set ends up using `/usr/lib64` correctly.

---

Also, re: the original question of what value to use for `CMAKE_SYSTEM_VERSION`:

>```
># CMAKE_SYSTEM_VERSION - on unix this is uname -r, for windows it is empty
>```

So this is actually the kernel version. I suppose we could set it to the output of `uname -r` ourselves, but for the stuff we're building the kernel version doesn't really make a difference. I'm leaning towards just leaving it as `1`